modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/provider/ObjectEntryInfoItemFormProvider.java

```diff
@@ -0,0 +1,201 @@
+
+/**
+ * @author Jorge Ferrer
+ */
+@Component(
+	immediate = true, property = Constants.SERVICE_RANKING + ":Integer=10",
+	service = InfoItemFormProvider.class
+)
+public class ObjectEntryInfoItemFormProvider
+	implements InfoItemFormProvider<ObjectEntry> {
+
+	@Override
+	public InfoForm getInfoForm() {
+		try {
+			return _getInfoForm(0);
+		}
+		catch (NoSuchFormVariationException noSuchFormVariationException) {
+			throw new RuntimeException(noSuchFormVariationException);
+		}
+	}
+
+	@Override
+	public InfoForm getInfoForm(ObjectEntry objectEntry) {
+		long objectDefinitionId = objectEntry.getObjectDefinitionId();
+
+		try {
+			return _getInfoForm(objectDefinitionId);
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(
+				StringBundler.concat(
+					"Unable to get object definition ", objectDefinitionId,
+					"for object entry ", objectEntry.getObjectEntryId()),
+				portalException);
+		}
+	}
+
+	@Override
+	public InfoForm getInfoForm(String formVariationKey)
+		throws NoSuchFormVariationException {
+
+		return _getInfoForm(GetterUtil.getLong(formVariationKey));
+	}
+
+	@Override
+	public InfoForm getInfoForm(String formVariationKey, long groupId)
+		throws NoSuchFormVariationException {
+
+		return _getInfoForm(GetterUtil.getLong(formVariationKey));
+	}
+
+	private InfoFieldSet _getBasicInformationInfoFieldSet() {
+		return InfoFieldSet.builder(
+		).infoFieldSetEntry(
+			ObjectEntryInfoItemFields.publishDateInfoField
+		).infoFieldSetEntry(
+			ObjectEntryInfoItemFields.authorNameInfoField
+		).infoFieldSetEntry(
+			ObjectEntryInfoItemFields.authorProfileImageInfoField
+		).labelInfoLocalizedValue(
+			InfoLocalizedValue.localize(getClass(), "basic-information")
+		).name(
+			"basic-information"
+		).build();
+	}
+
+	private InfoFieldSet _getDefinitionFieldSet(long objectDefinitionId)
+		throws NoSuchDefinitionException {
+
+		try {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.getObjectDefinition(
+					objectDefinitionId);
+
+			InfoLocalizedValue<String> fieldSetNameInfoLocalizedValue =
+				InfoLocalizedValue.singleValue(objectDefinition.getName());
+
+			return InfoFieldSet.builder(
+			).infoFieldSetEntry(
+				consumer -> {
+					for (ObjectField objectField :
+							_objectFieldLocalService.getObjectFields(
+								objectDefinitionId)) {
+
+						consumer.accept(
+							InfoFieldSet.builder(
+							).name(
+								objectField.getName()
+							).labelInfoLocalizedValue(
+								InfoLocalizedValue.singleValue(
+									objectField.getName())
+							).build());
+					}
+				}
+			).labelInfoLocalizedValue(
+				fieldSetNameInfoLocalizedValue
+			).name(
+				objectDefinition.getName()
+			).build();
+		}
+		catch (NoSuchDefinitionException noSuchDefinitionException) {
+			throw noSuchDefinitionException;
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException("Unexpected exception", portalException);
```

Comment:
Same as above. Maybe this is fine, but think it through

Reply:
Same as above.